third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -488,7 +532,8 @@ impl ValueImpl {
             | (Address(_), _)
             | (Container(_), _)
             | (ContainerRef(_), _)
-            | (IndexedRef(_), _) => {
+            | (IndexedRef(_), _)
+            | (DelayedFieldID { .. }, _) => {
```

Comment:
INTERNAL_TYPE_ERROR is a type of invariant violation. Invariant violations should not be used for user-triggerable errors.

Reply:
Actually above I compare delayed with delayed, which returns a runtime error. So this one is an invariant violation because delayed field should never be compared with something else (i.e. if replaced the value of an aggregator, we must have replaced values for all of them)